#!/usr/bin/env python

from dynamic_reconfigure.parameter_generator_catkin import *

# Parameter generator handle
gen = ParameterGenerator()

# Add double parameters for angular velocity control
# Name | Type | Level | Description | Default | Min | Max
gen.add("kp_angular", double_t, 0, "Proportional gain for angular velocity", 1.0, 0, 10)
gen.add("ki_angular", double_t, 0, "Integral gain for angular velocity", 0.1, 0, 5)
gen.add("kd_angular", double_t, 0, "Derivative gain for angular velocity", 0.0, 0, 1.0)

# Add double parameters for linear velocity control
gen.add("kp_linear", double_t, 0, "Proportional gain for linear velocity", 1.0, 0, 10)
gen.add("ki_linear", double_t, 0, "Integral gain for linear velocity", 0.0, 0, 5)
gen.add("kd_linear", double_t, 0, "Derivative gain for linear velocity", 0.0, 0, 1.0)

# Add boolean parameters for EKF sensor inputs
# Name | Type | Level | Description | Default
gen.add("use_imu", bool_t, 0, "Use IMU data", True)
gen.add("use_differential_drive_estimation", bool_t, 0, "Use Differential Drive Estimation data", True)
gen.add("use_vio", bool_t, 0, "Use VIO data", False)

# Exit parameter generator
# Package Name | Node Name (used to generate documentation only) | cfg File Name
exit(gen.generate("smb_control", "PIDControllerNode", "PID"))
